tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/aux/AuxOps.scala

```diff
@@ -61,4 +79,41 @@ object FunOps {
         state
     }
   }
+
+  /**
+   * Returns the expression: checkedElem \in setCell \land checkedElem = setElem
+   *
+   * @param rewriter
+   *   symbolic state rewriter
+   * @param checkedElem
+   *   element to be checked
+   * @param setElem
+   *   element to be compared against
+   * @param setCell
+   *   set to be compared against
+   * @param lazyEq
+   *   flag for use of lazy equality
+   * @return
+   *   a conjunction as described above
+   */
+  def inAndEq(
+      rewriter: SymbStateRewriter,
+      checkedElem: ArenaCell,
+      setElem: ArenaCell,
+      setCell: ArenaCell,
+      lazyEq: Boolean): TlaEx = {
+
+    val conjunction = if (lazyEq) {
+      tla.and(
+          tla.apalacheSelectInSet(checkedElem.toNameEx, setCell.toNameEx),
+          rewriter.lazyEq.cachedEq(checkedElem, setElem),
+      )
+    } else {
+      tla.and(
+          tla.apalacheSelectInSet(checkedElem.toNameEx, setCell.toNameEx),
+          tla.eql(checkedElem.toNameEx, setElem.toNameEx),
```

Comment:
```suggestion
          tla.apalacheSelectInSet(checkedElem.toBuilder, setCell.toBuilder),
          rewriter.lazyEq.cachedEq(checkedElem, setElem),
      )
    } else {
      tla.and(
          tla.apalacheSelectInSet(checkedElem.toBuilder, setCell.toBuilder),
          tla.eql(checkedElem.toBuilder, setElem.toBuilder),
```

Reply:
See my previous comment. This will lead to changes to `lazyEq` and the list goes on.